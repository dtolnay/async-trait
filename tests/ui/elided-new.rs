#![deny(bare_trait_objects)]

use async_trait::async_trait;

type Elided<'a> = &'a usize;

#[async_trait]
trait TestOk1 {
    async fn test_ok1(elided: Elided, okay: &usize);
}

#[async_trait]
trait TestOk2 {
    async fn test_ok2<'a>(elided: Elided<'a>, okay: &usize) -> &'a usize;
}

#[async_trait]
trait TestOk3 {
    async fn test_ok3(elided: Elided) -> &usize;
}

#[async_trait]
trait TestNok {
    async fn test_nok(elided: Elided, okay: &usize) -> &usize;
}

fn main() {}
