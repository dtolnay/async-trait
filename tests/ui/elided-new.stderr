error[E0106]: missing lifetime specifier
  --> $DIR/elided-new.rs:24:56
   |
24 |     async fn test_nok(elided: Elided, okay: &usize) -> &usize;
   |                               ------        ------     ^ expected named lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from argument 1 or argument 2
note: these named lifetimes are available to use
  --> $DIR/elided-new.rs:22:1
   |
22 | #[async_trait]
   | ^^^^^^^^^^^^^^
